da/celestia/celestia.go

```diff
@@ -111,34 +134,56 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *ty
 }
 
 // CheckBlockAvailability queries DA layer to check data availability of block at given height.
-func (c *DataAvailabilityLayerClient) CheckBlockAvailability(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultCheckBlock {
-	shares, err := c.client.NamespacedShares(ctx, c.namespaceID, dataLayerHeight)
-	code := dataRequestErrorToStatus(err)
-	if code != da.StatusSuccess {
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
+	header, err := c.rpc.Header.GetByHeight(ctx, dataLayerHeight)
```

Comment:
can you use blob module here also? `GetAll` and then check? instead of two calls?

Reply:
I think `SharesAvailable` is more efficient, as it doesn't need to return the whole blob data just to check that the block exists, even after two API calls cc @distractedm1nd

Reply:
Created follow up issue: https://github.com/rollkit/rollkit/issues/1040